### DISCUSSION TITLE
local meow key definition per state?

### DISCUSSION THREAD

**Discussion by Inc0n** [meow-edit/meow#256] [Q&A] [answered]
In evil `evil-define-key` can be used on a keymap, to define key binds on certain evil states for that map to be used.
It can be seen as local key definition, that doesn't affect the other keymaps/modes in similar states.
How can this be done in `meow`?

**Comment by Inc0n** [comment, marked as answer]
I found that a new state could be defined as shown in https://github.com/meow-edit/meow/blob/master/CUSTOMIZATIONS.org#meow-define-state. Which could be considered as a alternative, actually.

**Reply by astratagem** [reply]
I came to that conclusion, though I wonder if some solution to this use-case would be better off as a core Meow feature. It seems like a lot of boilerplate.

Perhaps https://github.com/emacs-evil/evil-collection may serve as a model for an implementation, and as an example of how many different modes need some kind of integration with `evil-mode`. Of course Meow is not Evil -- the integrations wouldn't need to be as extensive because Meow is not trying to emulate Vim.

**Reply by astratagem** [reply]
@Inc0n I was just checking out your emacs.d repo and it looks like you came up with some helpful macros!

https://github.com/Inc0n/emacs.d/blob/020c98a0ffbb14ea92ee3a54af43862ff24cc09f/lisp/obsolete/init-meow.el

Though it looks like you moved away from Meow, do you recall if those per-state macros worked out well for you?

**Reply by Inc0n** [reply]
Hi, yes I have moved onto using just emacs keybind.

I have defined `meow-define-key` is my cosmetic take on `meow-define-keys` as inspired by `evil-define-key`.

`my/meow-define-state-for-map` is a macro for simplifying that process. 

For the rest of the details, I can't remember any more details I am afraid. But you are more than welcome to use any code that may help :)

**Comment by DogLooksGood** [comment]
Meow is against to having its own keybinding system. The idea is to use keypad mode to transform user's existing keybindings.
